app/controllers/professors_controller.rb

```diff
@@ -12,8 +12,10 @@ def index
 
   # GET /professors/1
   # GET /professors/1.json
-  def avg_rev(course)
-    course.reviews.blank? ? 0 : course.reviews.average(:overall_rating).round(2)
+  def avg_rev(reviews)
+    sum = 0
+    reviews.each { |a| sum += a.overall_rating }
+    sum.to_f / reviews.length
   end
```

Comment:
This is identical code to `CoursesController#avg_rev`. Please refactor both functions into one shared function, that way modifying one modifies both.